### ISSUE TITLE
Being puzzled about the technique of `Fine-tune PWC-Net`

### ISSUE THREAD

**Issue by DelinQu** [GitCVfb/CVR#1]
Hi Bin Fan,
Thanks for your excellent repo! I'm a little puzzled about the technique of `Fine-tune PWC-Net` which is proposed in both **RSSR** and **CVR**:
- RSSR：
![image](https://user-images.githubusercontent.com/60593268/204069799-f798a842-7cc4-4889-8625-28b985a0ecec.png)
- CVR：
![image](https://user-images.githubusercontent.com/60593268/204069857-e8f09e6f-7a77-4dcb-8aeb-f45f9cd9399c.png)

It must be fine-tuned from PWC pre-trained model on the RS benchmark in a self-supervised way. But in [RSSR ](https://github.com/GitCVfb/RSSR/blob/cb0df6ba2769089097c61546b13066f41760f444/deep_unroll_net/model_RSSR.py#L67) and [CVR](https://github.com/GitCVfb/CVR/blob/0ec6c5f70a8d4434d0135f59258ebc7d70ba6a53/deep_unroll_net/model_CVR.py#L66), the Adam continued to jointly optimize PWC network with lr = 1e-5. I would like to ask if it is necessary to continue optimizing the PWC in RSSR/CVR after pwc-fine-tuning. Is it possible to freeze the PWC in RSSR/CVR? And, could you please provide the fine-tuned PWC weights?

Thanks in advance!

**Comment by GitCVfb** [comment]
Hello Delin,

Thank you for your interest in our work.

Optimizing PWC-Net together with the entire network helps to achieve better performance, which is compared in Table 1 of the RSSR's supplementary material (see Freezing or not freezing). We fine-tune PWC-Net on RS data according to the self-supervised training strategy inspired by https://github.com/ethliup/SelfDeblur. However, in my subsequent experimental experience, due to the joint overall training, this self-supervised fine-tuning of PWC-Net may have little improvement in the final recovery performance (e.g. LPIPS: 0.0697 vs. 0.0695, qualitative GS image quality). Another possible reason is that we focus on the final image generation rather than the quality of the optical flow. Note that training PWC-Net from scratch will significantly reduce metric performance. So I suggest that you can train RSSR directly based on the pre-trained PWC-Net model for convenience, which should be able to achieve the desired result. In addition, the pre-training model of NBMF in CVR is directly taken from RSSR.

Sincerely,
Fan Bin

**Comment by DelinQu** [comment]
> So I suggest that you can train RSSR directly based on the pre-trained PWC-Net model for convenience, which should be able to achieve the desired result.

Thank you so much for your rapid reply! 

I noticed [that](https://github.com/GitCVfb/CVR/blob/0ec6c5f70a8d4434d0135f59258ebc7d70ba6a53/train_carla.sh#L8) the model loaded in `CVR` is the jointly trained model `pretrained_rssr_net_flow` in `RSSR` instead of the `pre_pwc_net_flow` model which has been fine-tuned from PWC pre-trained on the RS benchmark. :D?

![image](https://user-images.githubusercontent.com/60593268/204201413-eeeb9250-46e3-4d41-ae8c-80dccdff6bbe.png)

![image](https://user-images.githubusercontent.com/60593268/204201305-76020c6d-ce9a-41c0-8cb4-5aec99194752.png)

I think the finetuned model is more recommended in CVR training, as it is difficult to figure out what happens to pwcnet during the joint training process of RSSR, and the pure fine-tuned `pwcnet` could be more convincing.  I'm a little confused about it.

**Comment by GitCVfb** [comment]
To train the CVR, directly based on the pre-trained RSSR (i.e. NBMF) is a reasonable choice. This is because the pre-trained RSSR (containing two modules) can provide better initial values, and subsequently the GS frame refinement module can better aggregate contextual information, etc. Essentially, the training of the CVR is based on the pre-trained full RSSR. Of course, it might be feasible to train the CVR directly based on a fine-tuned PWC-Net as you say, but I have not tried it. We used a two-stage training strategy instead. Moreover, since the second stage is joint training, the impact of doing or not doing fine-tuning would not be significant.

**Comment by DelinQu** [comment]
Thank you again for your reply! :D
